{
  "comments": [
    {
      "unresolved": true,
      "key": {
        "uuid": "68da2894_1fe6efc9",
        "filename": "src/main/java/com/googlesource/gerrit/plugins/replication/PushOne.java",
        "patchSetId": 1
      },
      "lineNbr": 393,
      "author": {
        "id": 1003883
      },
      "writtenOn": "2020-06-03T18:03:54Z",
      "side": 0,
      "message": "Does this change still output \"UnknownHostKey\" to the logs in this situation?",
      "range": {
        "startLine": 393,
        "startChar": 75,
        "endLine": 393,
        "endChar": 89
      },
      "revId": "583848945b150503fb76fb780f53bc5c2b42546c",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "20e1931e_c03181f4",
        "filename": "src/main/java/com/googlesource/gerrit/plugins/replication/PushOne.java",
        "patchSetId": 1
      },
      "lineNbr": 393,
      "author": {
        "id": 1011323
      },
      "writtenOn": "2020-06-03T18:37:52Z",
      "side": 0,
      "message": "Yes, with warning severity: [1]. As explained in the documentation there is an option to override the method handleKnownHostsFileUpdateFailure().\n\n* [1] https://github.com/apache/mina-sshd/blob/bb07cf5de38ac55a13fd9a6b91513a4ff6eebaec/sshd-core/src/main/java/org/apache/sshd/client/keyverifier/KnownHostsServerKeyVerifier.java#L618",
      "parentUuid": "68da2894_1fe6efc9",
      "range": {
        "startLine": 393,
        "startChar": 75,
        "endLine": 393,
        "endChar": 89
      },
      "revId": "583848945b150503fb76fb780f53bc5c2b42546c",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "9e33f4ae_5fb3e3e8",
        "filename": "src/main/java/com/googlesource/gerrit/plugins/replication/PushOne.java",
        "patchSetId": 1
      },
      "lineNbr": 393,
      "author": {
        "id": 1003883
      },
      "writtenOn": "2020-06-03T18:42:53Z",
      "side": 0,
      "message": "Great!",
      "parentUuid": "20e1931e_c03181f4",
      "range": {
        "startLine": 393,
        "startChar": 75,
        "endLine": 393,
        "endChar": 89
      },
      "revId": "583848945b150503fb76fb780f53bc5c2b42546c",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "5cd69380_2a780167",
        "filename": "src/main/java/com/googlesource/gerrit/plugins/replication/PushOne.java",
        "patchSetId": 1
      },
      "lineNbr": 393,
      "author": {
        "id": 1100310
      },
      "writtenOn": "2020-06-03T21:18:42Z",
      "side": 0,
      "message": "David, I\u0027m getting lost among all these small changes, and I don\u0027t know the Gerrit code well enough anymore to figure out the global picture. Best I can do is \"peephole reviews\".\n\nAnyway: if you use the org.eclipse.jgit.ssh.apache.SshdSessionFactory it\u0027ll use its own JGitServerKeyVerifier which won\u0027t go through the sshd KnownHostsServerKeyVerifier but use the configured ServerKeyDatabase. That one is by default its own OpenSshServerKeyDatabase. I don\u0027t think the exception message would contain the string \"UnknownHostKey\". It will still throw an exception, but the message is likely to be different. I think it\u0027d be \"Server key did not validate\".\n\nIf the knownhosts file cannot be updated, it also logs \"Could not update known hosts file\" with the filename and causing exception.",
      "parentUuid": "9e33f4ae_5fb3e3e8",
      "range": {
        "startLine": 393,
        "startChar": 75,
        "endLine": 393,
        "endChar": 89
      },
      "revId": "583848945b150503fb76fb780f53bc5c2b42546c",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "508e21f0_65c132c2",
        "filename": "src/main/java/com/googlesource/gerrit/plugins/replication/PushOne.java",
        "patchSetId": 1
      },
      "lineNbr": 393,
      "author": {
        "id": 1003883
      },
      "writtenOn": "2020-06-03T21:23:19Z",
      "side": 0,
      "message": "\u003e It will still throw an exception, but the message is likely to be different. I think it\u0027d be \"Server key did not validate\".\n\nThanks for the clarification. Although maybe a bit less explicit, that message probably still seems good enough for someone to understand the problem.",
      "parentUuid": "5cd69380_2a780167",
      "range": {
        "startLine": 393,
        "startChar": 75,
        "endLine": 393,
        "endChar": 89
      },
      "revId": "583848945b150503fb76fb780f53bc5c2b42546c",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "bc650fed_401d487f",
        "filename": "src/main/java/com/googlesource/gerrit/plugins/replication/PushOne.java",
        "patchSetId": 1
      },
      "lineNbr": 393,
      "author": {
        "id": 1011323
      },
      "writtenOn": "2020-06-04T06:02:02Z",
      "side": 0,
      "message": "Thanks, Thomas.",
      "parentUuid": "508e21f0_65c132c2",
      "range": {
        "startLine": 393,
        "startChar": 75,
        "endLine": 393,
        "endChar": 89
      },
      "revId": "583848945b150503fb76fb780f53bc5c2b42546c",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "bbed481d_db537dcb",
        "filename": "src/main/java/com/googlesource/gerrit/plugins/replication/PushOne.java",
        "patchSetId": 1
      },
      "lineNbr": 393,
      "author": {
        "id": 1024147
      },
      "writtenOn": "2021-11-30T18:35:38Z",
      "side": 0,
      "message": "Ack",
      "parentUuid": "bc650fed_401d487f",
      "range": {
        "startLine": 393,
        "startChar": 75,
        "endLine": 393,
        "endChar": 89
      },
      "revId": "583848945b150503fb76fb780f53bc5c2b42546c",
      "serverId": "173816e5-2b9a-37c3-8a2e-48639d4f1153"
    }
  ]
}